src/Api.js

```diff
@@ -33,6 +33,31 @@ class Api {
 		] );
 	}
 
+	/**
+	 * Get parsed edit summary for the given revision.
+	 * @param {number} revId
+	 * @return {Promise} Resolving Object with keys 'comment' and 'size'.
+ 	 */
+	fetchEditSummary( revId ) {
+		return this.mwApi.ajax( {
+			action: 'compare',
+			fromrev: revId,
+			torelative: 'prev',
+			prop: 'parsedcomment|size',
+			formatversion: 2
+		} ).then( data => {
+			if ( data.compare ) {
+				return $.Deferred().resolve( {
+					comment: data.compare.toparsedcomment,
+					size: data.compare.tosize - ( data.compare.fromsize || 0 )
+				} );
+			}
+			return $.Deferred().reject();
+		}, failData => {
+			return $.Deferred().reject( failData );
```

Comment:
Similarly, you don't need to return a rejected promise, because the promise is already rejecting at this point, just return failData directly.

Reply:
This doesn't work for me, if it's just

```
failData => {
    return failData;
}
```

If nothing's being done with `failData` then the whole error handler can just be left out, and it works correctly.

Reply:
Indeed! I've changed it back to returning a rejected promise.

`failData` is not being used, incidentally. I figure that's up to the caller of `fetchEditSummary()` to decide.